tools/clang/test/HLSLFileCheck/hlsl/objects/CbufferLegacy/double4_indexing.hlsl

```diff
@@ -0,0 +1,12 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
+
+// Make sure only 2 cbufferLoadLegacy.
+// CHECK:call %dx.types.CBufRet.f64 @dx.op.cbufferLoadLegacy.f64(i32 59, %dx.types.Handle %{{.*}}, i32 0)
+// CHECK:call %dx.types.CBufRet.f64 @dx.op.cbufferLoadLegacy.f64(i32 59, %dx.types.Handle %{{.*}}, i32 1)
+// CHECK-NOT:call %dx.types.CBufRet.f64
+
+double4 g;
+
+float main(int i:I) : SV_Target {
+  return g[i];
```

Comment:
While this produces two loads now, it could be optimized to one load with dynamic index into cb.  Therefore I propose updating the test to this:
```
double4 g;

float main(int2 i:I) : SV_Target {
  return g[i.x] + g[i.y];
}
```
Which will have to produce 2 loads no matter what optimization you do, and only produces 1 without the bug fix.